### ISSUE TITLE
Synchronous subscribe does not stop asynchronous events propagation

### ISSUE THREAD

**Issue by shushz** [Reactive-Extensions/RxJS#1065]
I'm observing a somewhat strange behavior and wondering whether it's me doing something wrong or is it possibly a bug.

TL;DR - While observable.subscribe is handling first (synchronous) event additional, asynchronous, events are being inserted into the stream, despite Node being in a blocking call. This causes weird issues, such as missed events or inability to dispose (as subscribe call does not return).

I'm developing an [atom](https://atom.io/) package with a [webview](http://electron.atom.io/docs/v0.30.0/api/web-view-tag/), which communicates with the main process over [IPC](http://electron.atom.io/docs/v0.30.0/api/ipc-renderer/). In one scenario a child process is spawned by the package on webview's request and its PID, ChildProcess events and stdout & stderr output are merged into an observable and sent back to the webview with its [send](http://electron.atom.io/docs/v0.30.0/api/web-view-tag/#lt-webview-gt-send-channel-args) command. I'm describing all of the parts, as I couldn't reproduce the issue with just a subset.

At the moment subscribe is performed there's only one synchronously created (with Observable.return) event in the stream. Other fromEvent() streams are merged into it and are waiting. All subscribe does, is forwards the event in onNext to the webview.

First weird behavior I observed was when I noticed that some events produced by ChildProcess, did not make it to the merged observable. Apparently since the observable that was supposed to pick them did not register in time. Although I found it strange to miss an event during a synchronous execution I decided to add a ReplaySubject as a kind of caching to make sure I get them.
It worked, - the events were no longer being missed. But, to my amazement, I saw that these asynchronously produced events were handled by the subscribe's onNext in the same first invocation as the one that has handled the first synchronous event. Effectively, since I have lots of events, this made the subscribe call never return. Thus I never got back the Disposable instance and the stream can't be safely disposed of.

I can workaround the problem by adding .delay() on the only synchronous observable and thus making the stream empty during the first subscribe call, but it felt really weird that I'd need to do that.

All of the schedulers are default for their operations. The software versions are:
rx: 3.1.1
Atom: 1.1.0
Electron: 0.30.7
IO.js: 2.3.1
OS X El Capitan 10.11.2

**Comment by justinwoo** [comment]
If `delay` helps it, I think that you might get the same fix by using a Scheduler. Could you try this?

**Comment by shushz** [comment]
Yes, using `Scheduler.default` works too. As is making the onNext handler asynchronous. That is not the main thing that made me wonder, however. What worries me and what I was trying to ask is: does it seem normal that events are propagating in the stream while the subscribe is blocked by a synchronous call? This, by the way, could just as easily be an issue introduced by the Electron's webview and not RX. Or maybe the combination of both. I'm not sure how to tell.
